### PR TITLE
Fixed test playbooks - roles filetree-{create,read} object-diff and cd

### PR DESCRIPTION
**Pull request by silvinux**
### What does this PR do?

1.- `roles  filetree-{create,read} object-diff` 

- added tag `always` to get the token in the test playbooks. 
- change localhost to all.

2.- playbooks/cd_gitlab_webhook_trigger.yml 

- Functionality is added in the CD playbook to process `Push events` or `Tag push events`. in gitlab, basically to work with tags and releases.
- Name changed to comply with collection naming convention

### How should this be tested?

It was tested manually.

### CODE REVIEW

File: playbooks/cd_gitlab_webhook_trigger.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+---
+- name: Gitlab Webhook Trigger Playbook
+  hosts: all
+  connection: local
+  gather_facts: false
+  tasks:
+    - name: "Get the modified files over the dirs from all the received commits"
+      set_fact:
```
Comment (ivarmu) [inline playbooks/cd_gitlab_webhook_trigger.yml]: This should be `ansible.builtin.set_fact`

File: playbooks/cd_gitlab_webhook_trigger.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        env: "{{ awx_webhook_payload.ref.split('/')[2] if awx_webhook_payload.ref.split('/')[1] == 'heads' else  awx_webhook_payload.project.default_branch }}"
```
Comment (ivarmu) [inline playbooks/cd_gitlab_webhook_trigger.yml]: Remove the extra space after the `else` keyword

File: playbooks/cd_gitlab_webhook_trigger.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        gitlab_scm_branch: "{{ awx_webhook_payload.ref.split('/')[2] }}"
+        orgs: "{{ awx_webhook_payload.project.name }}"
+        list_of_dirs: "{{ ((awx_webhook_payload.commits | map(attribute='added') | list) +
+                           (awx_webhook_payload.commits | map(attribute='modified') | list) +
+                           (awx_webhook_payload.commits | map(attribute='removed') | list))
+                       | flatten }}"
+
+    - set_fact:
```
Comment (ivarmu) [inline playbooks/cd_gitlab_webhook_trigger.yml]: As before, use FQCN

Comment (ivarmu) [inline playbooks/cd_gitlab_webhook_trigger.yml]: That task should be named

File: playbooks/cd_gitlab_webhook_trigger.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        regexpression: "/([^/]*)/env/(common|{{ env }})/controller_(.*).d/"
+
+    - name: "Get the Organization and the tags to run the CasC"
+      set_fact:
```
Comment (ivarmu) [inline playbooks/cd_gitlab_webhook_trigger.yml]: As before, user FQCN

File: playbooks/cd_gitlab_webhook_trigger.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        org_dirs_dict: "{{ (org_dirs_dict | default({})) | combine({input_var[0]: (((org_dirs_dict[input_var[0]] | default([])) + [input_var[1]] + ( ['projects','schedules'] if awx_webhook_payload.ref.split('/')[1] == 'tags' else [''] ) ) | unique | reject('match', '^$') ) }) }}"
```
Comment (ivarmu) [inline playbooks/cd_gitlab_webhook_trigger.yml]: Remove unneeded spaces

File: playbooks/cd_gitlab_webhook_trigger.yml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      vars:
+        input_var: "{{ item | regex_search(regexpression, '\\1', '\\3') }}"
+      loop: "{{ list_of_dirs }}"
+      when: input_var | type_debug is match('list')
+
+    - name: Configure Controller Job Launch | Launch launch_jobs Drop Diff (Delete)
+      include_role:
```
Comment (ivarmu) [inline playbooks/cd_gitlab_webhook_trigger.yml]: As before, use FQCN

File: playbooks/cd_gitlab_webhook_trigger.yml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            tags: "{{ org_tags.value }}"
+            wait: true
+            verbosity: 0
+      with_dict: "{{ org_dirs_dict }}"
+      loop_control:
+        loop_var: org_tags
+      when: org_dirs_dict is defined
+
+    - name: Configure Controller Job Launch | Launch launch_jobs creation
+      include_role:
```
Comment (ivarmu) [inline playbooks/cd_gitlab_webhook_trigger.yml]: As before, use FQCN

**Review by ivarmu — CHANGES_REQUESTED** [review]

File: playbooks/cd_gitlab_webhook_trigger.yml (hunk 3 above)
Comment (silvinux) [reply playbooks/cd_gitlab_webhook_trigger.yml]: Fixed. Thanks!

File: playbooks/cd_gitlab_webhook_trigger.yml (hunk 4 above)
Comment (silvinux) [reply playbooks/cd_gitlab_webhook_trigger.yml]: Fixed. Thanks!

File: playbooks/cd_gitlab_webhook_trigger.yml (hunk 6 above)
Comment (silvinux) [reply playbooks/cd_gitlab_webhook_trigger.yml]: Fixed. Thanks!

**Review by ivarmu — CHANGES_REQUESTED** [review]
The truthy values may be `false` or `true`

**Review by ivarmu — APPROVED** [review]

**Pull request merged**